### ISSUE TITLE
records: download all button

### ISSUE THREAD

**Issue by lnielsen** [zenodo/zenodo#210]

**Comment by lnielsen** [comment]
Can likely be achieved via AIP. Related to #34.

**Comment by fedorov** [comment]
+1

A possibility you could consider is for the user to "request download all" to initiate compression, and then send notification (by email) with a download link of the zip file when compression is done. You could make the link expire in 24 hours and then release the storage.

**Comment by slint** [comment]
Another solution might be to stream all the files inside a zip using a library such as https://github.com/SpiderOak/ZipStream. No extra temporary disk storage will be required and (hopefully, if the library works as advertised) no extra memory.

**Comment by peterjc** [comment]
I just uploaded a dataset of a few hundred files, and am shocked that there is no end user "download all button" giving a ZIP or tar-ball or similar (which could be generated on the fly).

In hindsight, I should have uploaded an archive myself, but the upload interface didn't give explicit guidance, and was clearly designed to cope with multiple files.

Is the current work around to instead upload a single archive (e.g. ``example.zip`` or ``example.tar.gz`` as appropriate for the targeted community), as a new version if necessary?

**Comment by lnielsen** [comment]
Yes, the current workaround is to upload a ZIP (better, and previewed - tar.gz is not previewed).

The problem here is that we have TB-sized datasets, and thus making a "download all" button is not trivial if it needs to scale.

**Comment by jcolomb** [comment]
An automatic zipping for smaller dataset (at the upload or even later) would then solve most problems, right? It would also save storage space ?

**Comment by zaneselvans** [comment]
I would also really appreciate this -- we have several files in [our archives](https://zenodo.org/record/3653159) so users have the option of grabbing only the data they need, but many folks want all of it, and it's tedious to have to click every single file. Or is the general expectation that folks just upload one big zip file as an archive?

**Comment by fgnievinski** [comment]
> Yes, the current workaround is to upload a ZIP (better, and previewed - tar.gz is not previewed).
> 
> The problem here is that we have TB-sized datasets, and thus making a "download all" button is not trivial if it needs to scale.

may I suggest including some guidance in the upload user interface. 

where it currently says:
```
Drag and drop files here
— or —
Choose files
(minimum 1 file required, max 50 GB per dataset - contact us for larger datasets)
```
There could be a note such as the following:

_Note: in case the dataset contains more than a few files, please consider packing them in a zipfile, to facilitate download by the user._

**Comment by fgnievinski** [comment]
btw, there is zenodo_get, a downloader for Zenodo records:

https://doi.org/10.5281/zenodo.1261812

**Comment by ivan-paleo** [comment]
> There could be a note such as the following:
> 
> _Note: in case the dataset contains more than a few files, please consider packing them in a zipfile, to facilitate download by the user._

It seems to me to be a good idea as long as there is no solution to "download all".

**Comment by ivan-paleo** [comment]
> 
> 
> btw, there is zenodo_get, a downloader for Zenodo records:
> 
> https://doi.org/10.5281/zenodo.1261812

What about integrating that tool into the Zenodo interface?

**Comment by ivan-paleo** [comment]
FYI, there is also an R package to do it (and many other things): https://github.com/eblondel/zen4R

**Comment by lnielsen** [comment]
Hi, it's not that we don't want to implement it. It's rather that its a task that likely takes ~2-3 weeks from start to launch and we're heavily resource constrained. As part of optimising resources, we're moving Zenodo on top of the InvenioRDM platform, so if you willing to help out, we're be more than happy to engage with you on how to implement it in InvenioRDM. We do have partners in InvenioRDM that's interested in the same feature, so any help would be much appreciated.
